File: ts/iq/optoutlist~set~item.ts
/** <iq type="set" xmlns="optoutlist">. Replies with one of: SuccessWithMatch, SuccessWithMismatch, InvalidRequest, ServerError. */
export function buildBlocklistsUpdateOptOutList(jid: string, category: string, action: string, dhash?: string, reason?: string, entryPoint?: string, signupId?: string, duration?: number): BinaryNode {
  return { tag: 'iq', attrs: { to: 's.whatsapp.net', xmlns: 'optoutlist', type: 'set' }, content: [
      { tag: 'item', attrs: { jid: jid, category: category, action: action, dhash: dhash, reason: reason, entry_point: entryPoint, signup_id: signupId, duration: duration } },
    ] }
}

File: rust/iq/optoutlist~set~item.rs
/// <iq type="set" xmlns="optoutlist">. Replies with one of: SuccessWithMatch, SuccessWithMismatch, InvalidRequest, ServerError.
pub fn build_blocklists_update_opt_out_list(jid: Jid, category: String, action: String, dhash: String, reason: String, entryPoint: String, signupId: String, duration: i64) -> Node {
    NodeBuilder::new("iq")
        .attr("to", "s.whatsapp.net")
        .attr("xmlns", "optoutlist")
        .attr("type", "set")
        .child(NodeBuilder::new("item").attr("jid", jid).attr("category", category).attr("action", action).attr("dhash", dhash).attr("reason", reason).attr("entry_point", entryPoint).attr("signup_id", signupId).attr("duration", duration).build())
        .build()
}